### PR TITLE
Add internal ContainerService and pod informer for kruise-daemon

### PR DESCRIPTION
**Pull request by FillZpp**
Signed-off-by: FillZpp <FillZpp.pub@gmail.com>

### Ⅰ. Describe what this PR does
Add internal ContainerService and pod informer for kruise-daemon.

The internal ContainerService is to get data that CRI does not support, such as environments of an existing container.

### II. Special notes for reviews
This is a basic (step 1) PR for #644 and #641 .

### CODE REVIEW

**Comment by Fei-Guo** [comment]
LGTM

File: pkg/daemon/criruntime/containerruntime/docker.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	client     *dockerapi.Client
+}
+
+func (d *dockerContainerService) createRuntimeClientIfNecessary() error {
+	d.Lock()
+	defer d.Unlock()
+	if d.client != nil {
+		return nil
+	}
+	c, err := dockerapi.NewClient(d.runtimeURI, "1.23", nil, nil)
```
Comment (wdwangcumt) [inline pkg/daemon/criruntime/containerruntime/docker.go:49]: `1.23`, the runtime version is hard coded, consider passing it via the `runtimeConfig` or another better way.

Comment (FillZpp) [reply pkg/daemon/criruntime/containerruntime/docker.go:49]: Thanks. Actually, this version indicates the version of the client (kruise-daemon), not the server (the docker-daemon running on node). So it is more related to how we use the client to get containers and images, and the server API should be compatible.

Comment (wdwangcumt) [reply pkg/daemon/criruntime/containerruntime/docker.go:49]: Oh, I got it, thanks very much!

File: pkg/features/kruise_features.go [hunk 2]
```diff
@@ -57,11 +57,16 @@ const (
...
+	// DaemonWatchingPod enables kruise-daemon to list watch pods that belong to the same node.
+	DaemonWatchingPod featuregate.Feature = "DaemonWatchingPod"
 )
 
 var defaultFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
-	PodWebhook:                       {Default: true, PreRelease: featuregate.Beta},
-	KruiseDaemon:                     {Default: true, PreRelease: featuregate.Beta},
+	PodWebhook:        {Default: true, PreRelease: featuregate.Beta},
+	KruiseDaemon:      {Default: true, PreRelease: featuregate.Beta},
+	DaemonWatchingPod: {Default: true, PreRelease: featuregate.Beta},
```
Comment (furykerry) [inline pkg/features/kruise_features.go:68]: gofmt again？

**Review by furykerry — APPROVED** [review]

**Pull request merged**